test/unit.js

```diff
@@ -0,0 +1,192 @@
+      .reply(404);
+
+    Check({ package: '../package.json', shrinkwrap: './npm-shrinkwrap.json' }, function (err) {
+
+      expect(err.message).to.equal('Got an invalid response from requireSafe, please email the above debug output to support@requiresafe.com');
+      done();
+    })
+;  });
+
+  it('Responds correctly to package being undefined', function (done) {
+
+    Check({ package: undefined, shrinkwrap: './npm-shrinkwrap.json' }, function (err) {
+
+      expect(err.message).to.equal('package.json is required');
+      done();
+    });
+  });
+
+  it('Responds correctly to both package and shrinkwrap being undefined', { timeout: 10000 }, function (done) {
+
+    Check({ package: undefined, shrinkwrap: undefined }, function (err) {
+
+      expect(err.message).to.equal('"value" must contain at least one of [package, shrinkwrap]');
+      done();
+    });
+  });
+
+  it('Responds correctly to receiving a 200 but no findings', function (done) {
+
+    Nock('https://api.requiresafe.com')
+      .post('/check')
+      .reply(200, []);
+
+    Check(workingOptions, function (err, results) {
+
+      expect(err).to.not.exist();
+      expect(results).to.deep.equal([]);
+      done();
+    });
+  });
+
+  it('Responds correctly to receiving a 200 and findings', function (done) {
+
+    Nock('https://api.requiresafe.com')
+      .post('/check')
+      .reply(200, findings);
+
+    Check(workingOptions, function (err, results) {
+
+      expect(err).to.not.exist();
+      expect(results).to.deep.include(findings);
+      done();
+    });
+  });
+
+  it('Handles package and shrinkwrap being objects', function (done) {
+
+    Nock('https://api.requiresafe.com')
+      .post('/check')
+      .reply(200, findings);
+
+    Check({ package: require(workingOptions.package), shrinkwrap: require(workingOptions.shrinkwrap) }, function (err, results) {
+
+      expect(err).to.not.exist();
+      expect(results).to.deep.include(findings);
+      done();
+    });
+  });
+
+  it('Responds correctly to a misc error', function (done) {
+
+    Nock('https://api.requiresafe.com')
+      .post('/check')
+      .replyWithError('Error: Client request error: connect ECONNREFUSED https://api.requiresafe.com');
+
+    Check(workingOptions, function (err, results) {
+
+      expect(err).to.exist();
+      expect(err.output.payload.message).to.equal('Error: Client request error: connect ECONNREFUSED https://api.requiresafe.com');
+      done();
+    });
+  });
+
+  it('Sends exceptions', function (done) {
+
+    var exceptions = ['https://requiresafe.com/advisories/39', 'https://requiresafe.com/advisories/9000'];
+
+    Nock('https://api.requiresafe.com')
+      .post('/check', JSON.stringify({
+        package: require(workingOptions.package),
+        shrinkwrap: require(workingOptions.shrinkwrap),
+        exceptions: exceptions
+      }))
+      .reply(200, findings);
+
+    Check(Object.assign(workingOptions, { exceptions: exceptions }), function (err, results) {
```

Comment:
`Object.assign()` doesn't exist in node 0.10, we need to keep these tests backwards compatible so you'll have to use something else.